rules/scripts/bitstreams_workspace.py

```diff
@@ -318,7 +319,7 @@ def main(argv):
     #   @bitstreams//:bitstream_mask_rom
     configured = cache.WriteBuildFile(args.build_file, bitstream)
     if args.bitstream != configured:
-        logging.error('Configured bitstream "{}" as {}.'.format(
+        logging.info('Configured bitstream "{}" as {}.'.format(
```

Comment:
If you want to quiet "latest" != configured here it may be worth a special case or a check. I at least want a warning if we're not running a specified bitstream.

Reply:
But this is not a warning as we **_are_** running the bitstream we specified. This case is hit anytime `args.bitstream = "latest"` because on line [262](https://github.com/lowRISC/opentitan/blob/26b2cabc4da62c4b675149910d8db160a4ecad82/rules/scripts/bitstreams_workspace.py#L262) we update the return value of `WriteBuildFile` with the actual commit hash that "latest" maps to.

If we passed in a commit hash to `args.bitstream` (instead of the default `"latest"` string) and this one wasn't found, then line [308](https://github.com/lowRISC/opentitan/blob/26b2cabc4da62c4b675149910d8db160a4ecad82/rules/scripts/bitstreams_workspace.py#L311) will log a warning now (as updated in this PR) if we can find a bitstream close enough to the one we wanted, or line [308](https://github.com/lowRISC/opentitan/blob/26b2cabc4da62c4b675149910d8db160a4ecad82/rules/scripts/bitstreams_workspace.py#L308) will log a warning and immediately kill the skip (which will result in a bazel build error).

Reply:
Please keep a check that can throw a warning at this stage in case someone messes up the logic at some point.

Reply:
Sorry, I'm a little confused here, I'm not sure I follow what you are asking. Are you asking to add a check or add a warning? If a check, what sort of check did you you have in mind (i.e., what expression would like me to check in addition to what is here)? (if a warning, I think line 311 is what you are looking for?)

Reply:
I don't think I've managed to express that to me, that line looks like it's a last check that we aren't quietly doing anything that could ruin my day and warn me about it. As we clean up warnings we should scope the cleanup narrowly and preserve this coverage. In the case you're experiencing with args.bitstream = "latest" I agree it shouldn't be a warning, but I want to keep the scope of that case narrow in case something is broken somewhere else.

I'd write
```
if args.bitstream != configured and args.bitstream != "latest":
        logging.warning('Configured bitstream "{}" as {}.'.format(
```
and think it's worth the extra complexity.

Reply:
And thanks for cleaning up warnings

Reply:
I did one better and took a pass to clean up naming and all the error checks (including adding ones). This should clear things up. Please let me know if you are confused, happy to walk you through this offline.